pkg/provider/provider.go

```diff
@@ -0,0 +1,604 @@
+package provider
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/containous/maesh/pkg/k8s"
+	"github.com/containous/maesh/pkg/topology"
+	"github.com/containous/traefik/v2/pkg/config/dynamic"
+	"github.com/sirupsen/logrus"
+)
+
+// TopologyBuilder is capable of building Topologies.
+type TopologyBuilder interface {
+	Build(ignored k8s.IgnoreWrapper) (*topology.Topology, error)
```

Comment:
A variable called `ignored`, looks like a red flag to me! Could you add a quick comment above this method to explain why this variable is named `ignored` here?

Reply:
We didn't think about it too much since it was named almost everywhere else. But that's right, `ignoredResources` would make more sense.